### ISSUE TITLE
Root should not be inside location block

### ISSUE THREAD

**Issue by schkovich** [voxpupuli/puppet-nginx#142]
Bad practice is currently enforced: ``` fail('Cannot create a location reference without a www_root [...])```. 

Instead option to add **www_root** should be removed and error raised if user tries to add root inside location block.

[Nginx Pitfalls: Root inside location block](http://wiki.nginx.org/Pitfalls#Root_inside_Location_Block)

**BAD:**
```
server {
    server_name www.domain.com;
    location / {
      root /var/www/nginx-default/;
      [...]
    }
}
```
**GOOD:**
```
server {
  server_name www.domain.com;
  root /var/www/nginx-default/;
  location / {
    [...]
  }
```

[Root moved to vhost_header](https://github.com/schkovich/puppet-nginx/commit/d22a25908460fae1d31f3047369db110a72a824b)

**Comment by schkovich** [comment]
solution #576

**Comment by khaefeli** [comment]
:thumbsup:

but I think #576 is a workaround. the module should do it correctly by default, to avoid the wrong behavior, if the DevOP is careless / inexperienced

**Comment by schkovich** [comment]
I agree. However see @jfryman [comment](https://github.com/jfryman/puppet-nginx/pull/143#issuecomment-36015661). I do understand his noble democratic desire to allow people doing stupidity but I still think that by default _root_ should be in server block. Case that user from whatever reasons still wants to place root in location block give her/him option to do it.

**Comment by khaefeli** [comment]
@schkovich I totally agree.

**Comment by jfryman** [comment]
@schkovich I think in that comment I said...

> I'd rather provide sane defaults and let a user make silly decisions on their own.

I do think this falls under a sane default. Throw up a PR, I'll review/merge.

**Comment by wyardley** [comment]
I have included a way to suppress logs in #888 
We do have some use cases that require setting different roots per location, but agree that the default behavior could be improved.
I'm going to close this one, just because it hasn't had any activity in ages, but, as they say, pull requests (with tests) will be reviewed and gratefully accepted if this is still an issue.
